lifting-state-up/lifting-state-up.md

```diff
@@ -0,0 +1,221 @@
+| `const updatedStudent = { ... }`                    | We create a variable, `updatedStudent`, to hold our updated student data.                                                                                                                                                                                      |
+| `id: props.id, ... emailData: props.email,`         | The key-value pairs in this object should match exactly what is required in `App`'s `updateStudentData` function. In that function, we're looking for the keys `id`, `nameData`, `emailData`, and `isPresentData`, to match the existing `studentData` format. |
+| `isPresentData: !props.isPresent`                   | When we click on the attendance button, the value of `isPresentData` should be the negation of `props.isPresent`.                                                                                                                                              |
+| `props.onUpdate(...);`                              | The value of `props.onUpdate` is a function. We invoke this function using parentheses `()`.                                                                                                                                                                   |
+| `updatedStudent`                                    | The function `props.onUpdate` ultimately accepts one argument, an updated student object. We can pass in our new `updatedStudent` object here.                                                                                                                 |
+
+## Attaching `onAttendanceButtonClick` to the Attendance Button
+
+Now that we have an event handler, let's ensure that we're _using_ this event handler!
+
+We need to attach our function to the attendance button's `onClick` attribute:
+
+<!-- prettier-ignore-start -->
+```js
+const Student = (props) => {
+
+    const onAttendanceButtonClick = () => {
+        const updatedStudent = {
+            id: props.id,
+            nameData: props.name,
+            emailData: props.email,
+            isPresentData: !props.isPresent
+        }
+
+        // Invoke the function passed in through the prop named "onUpate"
+        // This function is referenced by the name "updateStudent" in App
+        props.onUpdate(updatedStudent);
+    }
+
+    const nameColor = props.isPresent ? 'green' : 'red';
+
+    return (
+        <div>
+            <ul>
+                <li className={nameColor}>Nickname: {props.name}</li>
+                <li>Email: {props.email}</li>
+            </ul>
+            <button onClick={onAttendanceButtonClick}>Toggle if {props.name} is present</button>
+        </div>
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+## Verifying
+
+Let's see if our app works still!
```

Comment:
if → whether since we're interested in the state of the supposition
'still works' is more conversational. Also the adjacent sibilants work_s s_till don't sound great together.
```suggestion
Let's see whether our app still works!
```